docs/docs/development/new-behavior.md

```diff
@@ -0,0 +1,449 @@
+
+##### ZMK Event Manager
+
+Including `zmk/event_manager.h` is required for the following dependencies to function properly.
+
+- `zmk/events/position_state_changed.h`: Position events' state (on/off), source, position, and timestamps
+- `zmk/events/keycode_state_changed.h`: Keycode events' state (on/off), usage page, keycode value, modifiers, and timestamps
+- `zmk/events/modifiers_state_changed.h`: Modifier events' state (on/off) and modifier value
+
+Events can be used similarly to hardware interrupts, through the use of [listeners](#listeners-and-subscriptions).
+
+`return` values:
+
+- `ZMK_EV_EVENT_BUBBLE`: The function saw the event memory and is done using it. Keep propagating the event memory to the next listeners.
+- `ZMK_EV_EVENT_HANDLED`: The function received the event memory and has handled it. Stop propagating it to the next listeners and free the event payload memory.
+- `ZMK_EV_EVENT_CAPTURED`: The function now owns the event memory. Stop propagating it to other listeners. The event memory will be manually freed at the end of the function when we are done with it.
+
+Functions:
+
+- `ZMK_EVENT_RAISE(ev)`: Invoke event (`ev`) as soon as possible in the workqueue
+- `ZMK_EVENT_RAISE_AFTER(ev, mod)`: Invoke event (`ev`) as soon as possible in the workqueue, _after_ an event is captured by a [listener](#listeners-and-subscriptions) (`mod`)
+- `ZMK_EVENT_RAISE_AT(ev, mod)`: Invoke event (`ev`) as soon as possible in the workqueue, _at the same time_ an event is captured by a [listener](#listeners-and-subscriptions) (`mod`)
+- `ZMK_EVENT_RELEASE(ev)`: Release event (`ev`) as soon as possible
+- `ZMK_EVENT_FREE(ev)`: Free the memory associated with the event (`ev`)
+
+###### Listeners and Subscriptions
+
+The condensed form of lines 192-225 of the tap-dance driver, shown below, does an excellent job of showcasing the function of listeners and subscriptions with respect to the [ZMK Event Manager](#zmk-event-manager).
+
+```c title="app/src/behaviors/behavior_tap_dance.c (Lines 192-197, 225)"
+static int tap_dance_position_state_changed_listener(const zmk_event_t *eh);
+
+ZMK_LISTENER(behavior_tap_dance, tap_dance_position_state_changed_listener);
+ZMK_SUBSCRIPTION(behavior_tap_dance, zmk_position_state_changed);
+
+static int tap_dance_position_state_changed_listener(const zmk_event_t *eh){
+    // Do stuff...
+}
+```
+
+Listeners, defined by the `ZMK_LISTENER(mod, cb)` function, take in a listener name (`mod`) and a callback function (`cb`) as their parameters. On the other hand subscriptions are defined by the `ZMK_SUBSCRIPTION(mod, ev_type)`, and determine what kind of event (`ev_type`) should invoke the callback function from the listener. In the tap-dance example, this listener executes code depending on a `zmk_position_state_changed` event, or simply, a change in key position. Other types of ZMK events can be found as the name of the `struct` inside each of the files located at `app/include/zmk/events/<Event Type>.h`.
```

Comment:
Please note that the event listeners should return one of the `ZMK_EV_EVENT_*` values.